app/controllers/floating_ip_controller.rb

```diff
@@ -14,4 +14,267 @@ def self.display_methods
+    when "add"
+      @floating_ip = FloatingIp.new
+      options = form_params
+      ems = ExtManagementSystem.find(options[:ems_id])
+      options.delete(:ems_id)
+      task_id = ems.create_floating_ip_queue(session[:userid], options)
+
+      add_flash(_("Floating IP creation failed: Task start failed: ID [%{id}]") %
+                {:id => task_id.to_s}, :error) unless task_id.kind_of?(Fixnum)
+
+      if @flash_array
+        javascript_flash(:spinner_off => true)
+      else
+        initiate_wait_for_task(:task_id => task_id, :action => "create_finished")
+      end
+    end
+  end
+
+  def create_finished
+    task_id = session[:async][:params][:task_id]
+    floating_ip_name = session[:async][:params][:name]
+    task = MiqTask.find(task_id)
+    if MiqTask.status_ok?(task.status)
+      add_flash(_("Floating IP \"%{name}\" created") % { :name  => floating_ip_name })
+    else
+      add_flash(
+        _("Unable to create Floating IP \"%{name}\": %{details}") % { :name    => floating_ip_name,
+                                                                      :details => task.message }, :error)
+    end
+
+    @breadcrumbs.pop if @breadcrumbs
+    session[:edit] = nil
+    session[:flash_msgs] = @flash_array.dup if @flash_array
+
+    javascript_redirect :action => "show_list"
+  end
+
+  def delete_floating_ips
+    assert_privileges("floating_ip_delete")
+
+    floating_ips = if @lastaction == "show_list" || (@lastaction == "show" && @layout != "floating_ip")
+                     find_checked_items
+                   else
+                     [params[:id]]
+                   end
+
+    if floating_ips.empty?
+      add_flash(_("No Floating IPs were selected for deletion."), :error)
+    end
+
+    floating_ips_to_delete = []
+    floating_ips.each do |s|
+      floating_ip = FloatingIp.find(s)
+      if floating_ip.nil?
+        add_flash(_("Floating IP no longer exists."), :error)
+      else
+        floating_ips_to_delete.push(floating_ip)
+      end
+    end
+    process_floating_ips(floating_ips_to_delete, "destroy") unless floating_ips_to_delete.empty?
+
+    # refresh the list if applicable
+    if @lastaction == "show_list"
+      show_list
+      @refresh_partial = "layouts/gtl"
+    elsif @lastaction == "show" && @layout == "floating_ip"
+      @single_delete = true unless flash_errors?
+      if @flash_array.nil?
+        add_flash(_("The selected Floating IP was deleted"))
+      else # or (if we deleted what we were showing) we redirect to the listing
+        javascript_redirect :action => 'show_list', :flash_msg => @flash_array[0][:message]
+      end
+    end
+  end
+
+  def edit
+    assert_privileges("floating_ip_edit")
+    @floating_ip = find_by_id_filtered(FloatingIp, params[:id])
+    @in_a_form = true
+    drop_breadcrumb(
+      :name => _("Associate Floating IP \"%{name}\"") % { :name  => @floating_ip.name },
+      :url  => "/floating_ip/edit/#{@floating_ip.id}git")
```

Comment:
`/edit/#{@floating_ip.id}git`

I'm guessing `git` is not supposed to be there? :)